odxtools/database.py

```diff
@@ -125,13 +128,20 @@ def refresh(self) -> None:
         # let the diaglayers sort out the inherited objects and the
         # short name references
         for dlc in self.diag_layer_containers:
-            dlc._finalize_init(self._odxlinks)
+            dlc._finalize_init(self._odxlinks, self._ecu_shared_datas)
 
     @property
     def odxlinks(self) -> OdxLinkDatabase:
         """A map from odx_id to object"""
         return self._odxlinks
 
+    @property
+    def ecu_shared_datas(self) -> OdxLinkDatabase:
```

Comment:
that should return `NamedItemList[DiagLayer]` like the `.ecus` and `.protocols` properties. (on line 95 you define `self._ecu_shared_data` this way...)